### PR TITLE
Tier Discount

### PR DESCRIPTION
**Pull request by UtsavSomaiya**
- [ ] In this PR we can CRUD of discount Tier

### CODE REVIEW

File: admin/discount/add.php [hunk 1]
```diff
@@ -1,76 +1,126 @@
 <?php
     session_start();
     if (isset($_POST['submit'])) {
+        $discountTiers = [];
+        for ($i = 0; $i < sizeof($_POST['digit']); $i++) {
+            $newInput[$i] = [
+                    'minimum_spend_amount' => $_POST['minimum_spend_amount'][$i],
+                    'discount_digit' => $_POST['digit'][$i],
+                ];
```
Comment (hirenkeraliya) [inline admin/discount/add.php:9]: indentation always our first priority

File: admin/discount/add.php [hunk 2]
```diff
@@ -1,76 +1,126 @@
...
-        if ($count == 1) {
-            $_SESSION['digit_alert'] = "Already taken this discount";
-            $_SESSION['discount_name'] = $discountName;
-            $_SESSION['minimum_spend_amount'] =$minimumSpendAmount;
-            $_SESSION['digit'] = $discountDigit;
-            $_SESSION['type'] = $discountType;
-            $_SESSION['status'] = $discountStatus;
+        $discountDigits = [];
+        $minimumSpendAmounts = [];
+        for ($i = 0; $i < sizeof($_POST['digit']); $i++) {
```
Comment (hirenkeraliya) [inline admin/discount/add.php:60]: `sizeof` to `count`

File: admin/discount/add.php [hunk 3]
```diff
@@ -1,76 +1,126 @@
...
-        $insertDiscount->bindParam(':minimumAmount', $minimumSpendAmount);
         $insertDiscount->bindParam(':type', $discountType);
-        $insertDiscount->bindParam(':digit', $discountDigit);
         $insertDiscount->bindParam(':status', $discountStatus);
-        $isExecute = $insertDiscount->execute();
+        $insertDiscount->execute();
+        for ($i = 0; $i < sizeof($discountDigits); $i++) {
+            $insertTierDiscount = $pdo->prepare("INSERT INTO discount_tier(`discount_id`,`minimum_spend_amount`,`discount_digit`) SELECT max(`id`),$minimumSpendAmounts[$i],$discountDigits[$i] FROM `discount`");
+            $isExecute = $insertTierDiscount->execute();
+        }
```
Comment (hirenkeraliya) [inline admin/discount/add.php:116]: I think there is bug on the code

File: admin/discount/edit.php [hunk 4]
```diff
@@ -260,4 +414,8 @@
         </div>
     </div>
 </div>
+<script>
+    var index = <?= sizeof($discountTierIds) ?>;
```
Comment (hirenkeraliya) [inline admin/discount/edit.php:418]: totalDiscounts

File: admin/js/discount.js [hunk 5]
```diff
@@ -12,4 +12,50 @@ function discountStatusChanged(id, status) {
         },
         type: "POST"
     });
+}
+
+var j = 1;
```
Comment (hirenkeraliya) [inline admin/js/discount.js:17]: what is `j`?

Comment (hirenkeraliya) [reply admin/js/discount.js:17]: This mistake is repeated a second time and its last warning.

File: admin/js/discount.js [hunk 6]
```diff
@@ -12,4 +12,50 @@ function discountStatusChanged(id, status) {
...
+
+function addInsert() {
```
Comment (hirenkeraliya) [inline admin/js/discount.js:19]: make proper name what mean of `addInsert`?

File: admin/js/discount.js [hunk 7]
```diff
@@ -12,4 +12,50 @@ function discountStatusChanged(id, status) {
...
+    var remove = document.createElement('i');
+    remove.setAttribute('class', 'fa fa-trash-o');
+    remove.setAttribute('onclick', 'remove('+index+')');
+    document.getElementById('remove-div-' + index).appendChild(remove);
+
+    index++;
+}
+
+
+
```
Comment (hirenkeraliya) [inline admin/js/discount.js:56]: why this 3 blank lines?

**Review by hirenkeraliya — CHANGES_REQUESTED** [review]

**Review by hirenkeraliya — APPROVED** [review]

File: admin/discount/edit.php [hunk 8]
```diff
@@ -95,98 +189,48 @@
...
-                                        }
-                                        if (isset($discountType)) {
-                                            if ($discountType == "1") {
-                                                echo 'selected="selected"';
-                                            }
+                                    if (isset($_SESSION['type']) && $_SESSION['type'] == "1") {
+                                        echo 'selected="selected"';
+                                        unset($_SESSION['type']);
+                                    } else {
+                                        if ($discountType == "1") {
```
Comment (hirenkeraliya) [inline admin/discount/edit.php:222]: elseif

File: admin/discount/edit.php [hunk 9]
```diff
@@ -95,98 +189,48 @@
...
-                                                echo 'selected="selected"';
-                                                unset($_SESSION['type']);
-                                            }
-                                        }
-                                        if (isset($discountType)) {
+                                        if (isset($_SESSION['type']) && $_SESSION['type'] == "2") {
+                                            echo 'selected="selected"';
+                                            unset($_SESSION['type']);
+                                        } else {
                                             if ($discountType == "2") {
```
Comment (hirenkeraliya) [inline admin/discount/edit.php:234]: elseif

File: admin/layout/db_connect.php [hunk 10]
```diff
@@ -1,6 +1,8 @@
 <?php
     try {
         $pdo = new PDO('mysql:host=127.0.0.1;dbname=point-of-sale', 'root', '1234');
+        $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+        $pdo->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
```
Comment (hirenkeraliya) [inline admin/layout/db_connect.php:5]: ?

File: admin/product/add_product.php [hunk 11]
```diff
@@ -46,13 +46,6 @@
         $productStock = $_POST['product_stock'];
         $_SESSION['product_stock'] = $productStock;
 
-        $productImage = $_FILES['product_image']['name'];
-        $imageExtension = pathinfo($productImage, PATHINFO_EXTENSION);
-        if ($imageExtension != "jpg" && $imageExtension != "png" && $imageExtension != "jpeg" && $imageExtension != "gif") {
-            $_SESSION['file_alert'] = "Sorry, only JPG, JPEG, PNG & GIF files are allowed.";
-            header('location:../product/add_product.php');
-            exit;
-        }
```
Comment (hirenkeraliya) [inline admin/product/add_product.php:55]: why?

File: custom.js [hunk 12]
```diff
@@ -269,7 +285,7 @@ function toast() {
     }, 3000);
 }
 
-function discountApply(count) {
+/* function discountApply(count) {
```
Comment (hirenkeraliya) [inline custom.js:288]: why comment?

File: discount.php [hunk 13]
```diff
@@ -41,15 +47,15 @@
...
                                 </td>
-                                <td>
-                                    <button class="bg-red-500 text-white font-bold py-2 px-4 rounded-full" onclick="discountApply(<?= $count ?>)" id="discount-button-<?= $count ?>">
+                                <!-- <td>
+                                    <button class="bg-red-500 text-white font-bold py-2 px-4 rounded-full" id="discount-button-<?= $count ?>">
                                         Apply
                                     </button>
-                                </td>
+                                </td> -->
                             </tr>
```
Comment (hirenkeraliya) [inline discount.php:59]: why commented code

**Review by hirenkeraliya — CHANGES_REQUESTED** [review]

File: admin/layout/db_connect.php (hunk 10 above)
Comment (UtsavSomaiya) [reply admin/layout/db_connect.php:5]: This is for show the pdo exception.

**Review by hirenkeraliya — APPROVED** [review]

**Pull request merged**